src/elf.rs

```diff
@@ -1026,58 +913,26 @@ impl<C: ContextObject> Executable<C> {
                 };
                 let key = function_registry.register_function_hashed_legacy(
                     loader,
-                    !sbpf_version.static_syscalls(),
+                    true,
                     name.as_bytes(),
                     target_pc as usize,
                 )?;
-                if !sbpf_version.static_syscalls() {
-                    let offset = i.saturating_mul(ebpf::INSN_SIZE).saturating_add(4);
-                    let checked_slice = text_bytes
-                        .get_mut(offset..offset.saturating_add(4))
-                        .ok_or(ElfError::ValueOutOfBounds)?;
-                    LittleEndian::write_u32(checked_slice, key);
-                }
+                let offset = i.saturating_mul(ebpf::INSN_SIZE).saturating_add(4);
+                let checked_slice = text_bytes
+                    .get_mut(offset..offset.saturating_add(4))
+                    .ok_or(ElfError::ValueOutOfBounds)?;
+                LittleEndian::write_u32(checked_slice, key);
             }
         }
 
-        let mut program_header: Option<&Elf64Phdr> = None;
-
         // Fixup all the relocations in the relocation section if exists
         for relocation in elf.dynamic_relocations_table().unwrap_or_default().iter() {
-            let mut r_offset = relocation.r_offset as usize;
-
-            // When sbpf_version.enable_elf_vaddr()=true, we allow section.sh_addr !=
-            // section.sh_offset so we need to bring r_offset to the correct
-            // byte offset.
-            if sbpf_version.enable_elf_vaddr() {
-                match program_header {
-                    Some(header) if header.vm_range().contains(&(r_offset as u64)) => {}
-                    _ => {
-                        program_header = elf
-                            .program_header_table()
-                            .iter()
-                            .find(|header| header.vm_range().contains(&(r_offset as u64)))
-                    }
-                }
-                let header = program_header.as_ref().ok_or(ElfError::ValueOutOfBounds)?;
-                r_offset = r_offset
-                    .saturating_sub(header.p_vaddr as usize)
-                    .saturating_add(header.p_offset as usize);
-            }
+            let r_offset = relocation.r_offset as usize;
 
             match BpfRelocationType::from_x86_relocation_type(relocation.r_type()) {
                 Some(BpfRelocationType::R_Bpf_64_64) => {
                     // Offset of the immediate field
-                    let imm_offset = if text_section
-                        .file_range()
-                        .unwrap_or_default()
-                        .contains(&r_offset)
```

Comment:
Shouldn't we keep checking the if the offset is contained in the range for these two cases?